### ISSUE TITLE
Problems identifying the correct character formatting from docx document

### ISSUE THREAD

**Issue by mr-bill** [jgm/pandoc#6499]
I am having many problems recovering the correct format of the text.
Problems arise when document styles (character styles and/or paragraph styles) contain information about character formatting.
(_This is a common practice in publishing to be able to "tag" documents with styles that can then be used in XML databases._)

First case (paragraph styles):
As you can see in the attached example file, paragraphs 1 to 4 are visually the same as those from 5 to 8. 
"`pandoc -f docx -t html test_styles.docx -o test.html`": the html/markdown is **different**.
"`pandoc -f docx+style -t html test_styles.docx -o test.html`": the html/markdown is **different**.

Second case (character styles)
Paragraphs 9 to 12 are the same as paragraphs 13 to 16.
"`pandoc -f docx`": the html/markdown is the **same**.
"`pandoc -f docx+style -t html test_styles.docx -o test.html`": the html/markdown is **different**.

**Is it a bug?**
Because it wouldn't make sense as "desired behavior".
Is there a **patch** or **workaround**?

Thank you very much

[test_styles.docx](https://github.com/jgm/pandoc/files/4861281/test_styles.docx)

**Comment by mb21** [comment]
What do you mean with "character formatting"? Can you post the actual and expected html output?

Also please make sure to use the newest pandoc version.

**Comment by tarleb** [comment]
The problem is probably that the commands don't include the `-s` or `--standalone` flag. This will add the necessary tags to mark the content encoding as UTF-8. Otherwise the browser will guess, and will often guess wrong.

**Comment by mr-bill** [comment]
Hi
Character formatting is the bold, italic, bold-italic style...

This is an extract from text/doc:

> “MyParaNormal” para style: regular, <em>italic</em>, <strong>bold</strong>, <em><strong>bold-italic</strong></em>.
> <em>“MyParaItalic” para style: </em>regular<em>, italic, </em><strong>bold</strong><em>, <strong>bold-italic</strong>.</em>
> <strong>“MyParaBold” para style: </strong>regular<strong>, </strong><em>italic</em><strong>, bold, <em>bold-italic</em>.</strong>
> <em><strong>“MyParaBoldItalic” para style: </strong></em>regular<em><strong>, </strong>italic<strong>, </strong></em><strong>bold<em>, bold-italic.</em></strong>
> 
> “default” para-style, “default” char-style: regular, <em>italic</em>, <strong>bold</strong>, <em><strong>bold-italic</strong></em>.
> <em>“default” para-style, “MyCharItalic” char-style: </em>regular<em>, italic, </em><strong>bold</strong><em>, <strong>bold-italic</strong>.</em>
> <strong>“default” para-style, “MyCharBold” char-style: </strong>regular<strong>, </strong><em>italic</em><strong>, bold, <em>bold-italic</em>.</strong>
> <em><strong>“default” para-style, “MyCharBoldItalic” char-style: </strong></em>regular<em><strong>, </strong>italic<strong>, </strong></em><strong>bold<em>, bold-italic.</em></strong>


Html expected:

```
“MyParaNormal” para style: regular, <em>italic</em>, <strong>bold</strong>, <em><strong>bold-italic</strong></em>.
<em>“MyParaItalic” para style: </em>regular<em>, italic, </em><strong>bold</strong><em>, <strong>bold-italic</strong>.</em>
<strong>“MyParaBold” para style: </strong>regular<strong>, </strong><em>italic</em><strong>, bold, <em>bold-italic</em>.</strong>
<em><strong>“MyParaBoldItalic” para style: </strong></em>regular<em><strong>, </strong>italic<strong>, </strong></em><strong>bold<em>, bold-italic.</em></strong>

“default” para-style, “default” char-style: regular, <em>italic</em>, <strong>bold</strong>, <em><strong>bold-italic</strong></em>.
<em>“default” para-style, “MyCharItalic” char-style: </em>regular<em>, italic, </em><strong>bold</strong><em>, <strong>bold-italic</strong>.</em>
<strong>“default” para-style, “MyCharBold” char-style: </strong>regular<strong>, </strong><em>italic</em><strong>, bold, <em>bold-italic</em>.</strong>
<em><strong>“default” para-style, “MyCharBoldItalic” char-style: </strong></em>regular<em><strong>, </strong>italic<strong>, </strong></em><strong>bold<em>, bold-italic.</em></strong>
```

**Comment by mb21** [comment]
what's you pandoc version?

**Comment by mr-bill** [comment]
> The problem is probably that the commands don't include the `-s` or `--standalone` flag. This will add the necessary tags to mark the content encoding as UTF-8. Otherwise the browser will guess, and will often guess wrong.

No. Is not a problem of encoding in output html. (However I did the test with "-s", thanks. ;) )
It is a problem of reading and correct recognition of the style.

It is like this #6452 
I suspect there are several ways for Word (but also LibreOffice) to write the same visual result of formatting.

**Comment by mr-bill** [comment]
> what's your pandoc version?

pandoc 2.9.2.1
Compiled with pandoc-types 1.20, texmath 0.12.0.1, skylighting 0.8.3.2

**Comment by mb21** [comment]
okay, will have to unzip the docx to see what specific XML tags are in there that would indicate bold/italics that we do not parse...

**Comment by agusmba** [comment]
> Problems arise when document styles (character styles and/or paragraph styles) contain information about character formatting.

AFAIK pandoc does not support extracting formatting attributes (italic, bold, etc.) from docx's styles. It only extracts the *name* of the custom-style so you can process your sources accordingly if you need.

**Comment by jgm** [comment]
@jkr any thoughts on this?

**Comment by mr-bill** [comment]
> AFAIK pandoc does not support extracting formatting attributes (italic, bold, etc.) from docx's styles. It only extracts the _name_ of the custom-style so you can process your sources accordingly if you need.
> 
> It does extract _italic_, or **bold** when they are explicitly set on the text, just not via styles

But in this way there are anomalous behaviors that do not allow a correct conversion (and not because of a lack of expressiveness of the markdown language), a.e.:

> <em>“MyParaItalic” para style: </em>regular<em>, italic, </em><strong>bold</strong><em>, <strong>bold-italic</strong>.</em>

It should be converted to
`<div data-custom-style="MyParaItalic"><p><em>“MyParaItalic” para style: </em>regular<em>, italic, </em><strong>bold</strong><em>, <strong>bold-italic</strong>.</em></p></div>`

Instead we get
`<div data-custom-style="MyParaItalic"><p>“MyParaItalic” para style: regular, italic, <strong>bold</strong>, <strong>bold-italic</strong>.</p></div>`

_Regular_ and _Italic_ become the **_same thing_**!
(without possibility of posterior fixes)

**Comment by mr-bill** [comment]
> AFAIK pandoc does not support extracting formatting attributes (italic, bold, etc.) from docx's styles. It only extracts the _name_ of the custom-style so you can process your sources accordingly if you need.
> 
> It does extract _italic_, or **bold** when they are explicitly set on the text, just not via styles

**Also**: character styles are managed in _some way_, only paragraph styles seem to be ignored entirely.
In fact if we have:

> <em>“default” para-style, “MyCharItalic” char-style: </em>regular<em>, italic, </em><strong>bold</strong><em>, <strong>bold-italic</strong>.</em>

With "pandoc -f docx+styles -t html test_styles.docx -o test.html" we have
`<div data-custom-style="Normal"><p><span data-custom-style="MyCharItalic">“default” para-style, “MyCharItalic” char-style: regular, italic, <strong>bold</strong>, <strong>bold-italic</strong>.</span></p></div>`

That it is wrong.
(Same problem as before)

But if we use "pandoc -f docx -t html test_styles.docx -o test.html" we have
`<p><em>“default” para-style, “MyCharItalic” char-style:</em> regular<em>, italic,</em> <strong>bold</strong><em>, <strong>bold-italic</strong>.</em></p>`

That it is **right**!
